tests/vmi_networking_test.go

```diff
@@ -791,6 +793,129 @@ var _ = Describe("[Serial][rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][le
 				}, 120*time.Second).Should(Succeed())
 			})
 		})
+
+		Context("MTU verification", func() {
+			var vmi *v1.VirtualMachineInstance
+			var anotherVmi *v1.VirtualMachineInstance
+
+			getMtu := func(pod *k8sv1.Pod, ifaceName string) int {
+				output, err := tests.ExecuteCommandOnPod(
+					virtClient,
+					pod,
+					"compute",
+					[]string{"cat", fmt.Sprintf("/sys/class/net/%s/mtu", ifaceName)},
+				)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+				output = strings.TrimSuffix(output, "\n")
+				mtu, err := strconv.Atoi(output)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				return mtu
+			}
+
+			BeforeEach(func() {
+				var err error
+
+				By("Create VMI")
+				vmi = libvmi.NewFedora()
+
+				vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Create another VMI")
+				anotherVmi = masqueradeVMI([]v1.Port{}, "")
+				anotherVmi, err = virtClient.VirtualMachineInstance(anotherVmi.Namespace).Create(anotherVmi)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Wait for VMIs to be ready")
+				vmiReady := make(chan bool)
+				go func() {
+					tests.WaitUntilVMIReady(vmi, tests.LoggedInFedoraExpecter)
+					vmiReady <- true
+				}()
+
+				anotherVmiReady := make(chan bool)
+				go func() {
+					tests.WaitUntilVMIReady(anotherVmi, tests.LoggedInCirrosExpecter)
+					anotherVmiReady <- true
+				}()
```

Comment:
You understood it correctly but I think this is an overkill.
Waiting in sequence is enough, no need for goroutine here.
After you wait for the 1st, the other is either already ready or we need a few more seconds. If everything goes right, we will not wait more than needed.

Except that, using a goroutine is a problem when the helper functions assert, ginkgo do not like it and cause problems. It requires special handling.